drivers/backup/portworx/portworx.go

```diff
@@ -1535,6 +1617,166 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
+	var (
+		preUid             string
+		preRuleName        string
+		postRuleName       string
+		postUid            string
+		preActionValue     []string
+		preContainer       []string
+		postActionValue    []string
+		postContainer      []string
+		postBackground     []bool
+		postRunInSinglePod []bool
+		preBackground      []bool
+		preRunInSinglePod  []bool
+		preRulesInfo       api.RulesInfo
+		postRulesInfo      api.RulesInfo
+		prePodSelector     []map[string]string
+		postPodSelector    []map[string]string
+	)
+	if len(appParameters) == 0 {
+		appParameters = append(appParameters, AppRuleMaster)
+	}
+
+	for i := 0; i < len(appList); i++ {
+		if reflect.DeepEqual(appParameters[0][appList[i]].PreRule, backup.PreRule{}) {
+			log.Infof("Pre rule not required for application %v", appList[i])
+		} else {
+			for j := 0; j < len(appParameters[0][appList[i]].PreRule.Rule.PodSelectorList); j++ {
+				ps := strings.Split(appParameters[0][appList[i]].PreRule.Rule.PodSelectorList[j], "=")
+				psMap := make(map[string]string)
+				psMap[ps[0]] = ps[1]
+				prePodSelector = append(prePodSelector, psMap)
+				preActionValue = append(preActionValue, appParameters[0][appList[i]].PreRule.Rule.ActionList[j])
+				backgroundVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PreRule.Rule.Background[j])
+				preBackground = append(preBackground, backgroundVal)
+				podVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PreRule.Rule.RunInSinglePod[j])
+				preRunInSinglePod = append(preRunInSinglePod, podVal)
+				containerName := fmt.Sprintf("%s-%s", "container", appList[i])
+				preContainer = append(preContainer, os.Getenv(containerName))
+			}
+		}
+
+		if reflect.DeepEqual(appParameters[0][appList[i]].PostRule, backup.PostRule{}) {
+			log.Infof("Post rule not required for application %v", appList[i])
+		} else {
+			for j := 0; j < len(appParameters[0][appList[i]].PostRule.Rule.PodSelectorList); j++ {
+				ps := strings.Split(appParameters[0][appList[i]].PostRule.Rule.PodSelectorList[j], "=")
+				psMap := make(map[string]string)
+				psMap[ps[0]] = ps[1]
+				postPodSelector = append(postPodSelector, psMap)
+				postActionValue = append(postActionValue, appParameters[0][appList[i]].PostRule.Rule.ActionList[j])
+				backgroundVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PostRule.Rule.Background[j])
+				postBackground = append(postBackground, backgroundVal)
+				podVal, _ := strconv.ParseBool(appParameters[0][appList[i]].PostRule.Rule.RunInSinglePod[j])
+				postRunInSinglePod = append(postRunInSinglePod, podVal)
+				containerName := fmt.Sprintf("%s-%s", "container", appList[i])
+				postContainer = append(postContainer, os.Getenv(containerName))
+			}
+		}
+
+	}
+	totalPreRules := len(preActionValue)
+	totalPostRules := len(postActionValue)
+
+	if totalPreRules != 0 {
+		preRuleName = fmt.Sprintf("pre-rule-%v", time.Now().Unix())
```

Comment:
Can we change all `time.Now().Unix()` to generating random chars?

Reply:
done.
Cannot use "RandomString" from common.go as it would create "import cycle not allowed" error

Reply:
Then what we are doing here is incorrect. This function should be part of `backup_helper.go`.
Can we move the function and AppParameters? There is no reason it should be in `portworx.go`

Reply:
done.